File: Competitive Programing/Extreme/C++/War_games/War_games/War_games.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

const int card_values[14] = { 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14 };


string champion(vector<int>& man1, vector<int>& man2) {
    int current_index = 0;

    while (!man1.empty() && !man2.empty()) {

        int player1_card = man1[current_index];
        int player2_card = man2[current_index];

        if (player1_card < player2_card) {
           
            man2.push_back(player1_card);
            man2.push_back(player2_card);
            man1.erase(man1.begin() + current_index);
            man2.erase(man2.begin() + current_index);

        }
        else if (player1_card > player2_card) {
            
            man1.push_back(player2_card);
            man1.push_back(player1_card);
            man2.erase(man2.begin() + current_index);
            man1.erase(man1.begin() + current_index);

        }
        else if (player1_card == player2_card) {
           
            man1.push_back(player1_card);
            man2.push_back(player2_card);
            man1.erase(man1.begin() + current_index);
            man2.erase(man2.begin() + current_index);

        }
    }

    if (man1.empty()) {
        return "player 2";

    }
    else if (man2.empty()) {
      
        return "player 1";

    }
    else {
       
        return "draw";

    }
}

int main() {
    int n;
    cin >> n;

 
    for (int i = 0; i < n; i++) {

        vector<int> man1, man2;

        
        for (int j = 0; j < 52; j++) {
            int card;
            cin >> card;
            man1.push_back(card_values[card - 1]);
        }

        for (int j = 0; j < 52; j++) {
            int card;
            cin >> card;
            man2.push_back(card_values[card - 1]);
        }

        string champion1 = champion(man1, man2);

      
        cout << champion1 << endl;

    }

    return 0;
}

File: Competitive Programing/Extreme/War Games.py
n = int(input())

def champion(man1, man2):
    l = {"2": 2, "3": 3, "4": 4, "5": 5, "6": 6, "7": 7, "8": 8, "9": 9, "T": 10, "J": 11, "Q": 12, "K": 13, "A": 14}
    man1 = [l[card] for card in man1]
    man2 = [l[card] for card in man2]
    yyy=0
    while man1!=[] and man2!=[]:
        p=man1[yyy]
        Q=man2[yyy]
        if p<Q:
            man2.append(p)
            man2.pop(yyy)
            man1.pop(yyy)

        elif p>Q:
            man2.pop(yyy)
            man1.append(Q)
            man1.pop(yyy)
        elif p==Q:
            man1.append(p)
            man1.pop(yyy)
            man2.append(Q)
            man2.pop(yyy)
        if man1==man2:
            break
    if man1==[]:
        return'player 2'
    elif man2==[]:
        return 'player 1'
    else:
        return "draw"
ans=[]
for _ in range(0,n):
    l1=list(map(str,input().strip().split()))
    l2=list(map(str,input().strip().split()))
    print(champion(l1, l2))